### PR TITLE
Support Get<T> & Set<T> in ISessionCollection

### PR DESCRIPTION
**Pull request by hishamco**
This PR made for the issue #3 
/cc @davidfowl @Eilon

### CODE REVIEW

File: samples/SessionSample/BooleanFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+using System;
+using Microsoft.AspNet.Http;
+
+namespace SessionSample
+{
+    public class BooleanFormatter:ISessionFormatter<bool>
```
Comment (Tratcher) [inline samples/SessionSample/BooleanFormatter.cs]: Spacing

**Comment by Tratcher** [comment]
Would it be better to put these over here with the others? https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Http.Extensions/SessionCollectionExtensions.cs

**Comment by hishamco** [comment]
@Tratcher I will try to make a PR in HttpAbstraction repo

**Comment by davidfowl** [comment]
@hishamco Assuming this is just a design pull request, we don't need to harp on the formatting issues right now but please read the coding guidelines as well (https://github.com/aspnet/Home/blob/master/CONTRIBUTING.md).

I think @Tratcher has a point about where this should live. Also who is the one that needs to get to it? Is it always passed in? Is it registered in the service collection? 

I would be great if this could be registered as an open generic service with something like bond or protobuf:

```C#
services.AddScoped(typeof(ISessionFormatter<>), typeof(ProtobufSessionFormatter<>));
```

Then you could consume this like so:

```C#
public class MyController : Controller
{
    private ISessionFormatter<MyComplexType> _formatter;
    public MyController(ISessionFormatter<MyComplexType> formatter)
    {
        _formatter = formatter;
    }

    [HttpPost("/")]
    public void DoSomething(MyType type)
    {
          Context.Session.Set(myType, _formatter);
    }
}
```

**Comment by hishamco** [comment]
@davidfowl thanks for your comments

**Comment by hishamco** [comment]
@davidfowl I agreed to be register in the service collection, but why we need protobuf if the developer explicitly will implement ISessionFormatter

**Pull request closed without merging**